pkg/server/api/svid/v1/server_test.go

```diff
@@ -146,13 +147,119 @@ func TestMintX509SVID(t *testing.T) {
+	defer c.Close()
+
+	spiffeID := spiffeid.Must("example.org", "workload1")
+
+	testCases := []struct {
+		name string
+
+		req        *svidpb.MintJWTSVIDRequest
+		err        string
+		code       codes.Code
+		issuedAt   time.Time
+		expiresAt  time.Time
+		logMsg     string
+		serviceErr string
+		token      string
+		spiffeID   spiffeid.ID
+	}{
+		{
+			name:      "success",
+			spiffeID:  spiffeID,
+			token:     "token",
+			issuedAt:  time.Now(),
+			expiresAt: time.Now().Add(time.Minute),
+			req: &svidpb.MintJWTSVIDRequest{
+				Id: &types.SPIFFEID{
+					TrustDomain: spiffeID.TrustDomain().String(),
+					Path:        spiffeID.Path(),
+				},
+				Audience: []string{"audience1", "audience2"},
+				Ttl:      60,
+			},
+		},
+		{
+			name:   "request missing SPIFFE ID",
+			err:    "request must specify SPIFFE ID",
+			logMsg: "Request must specify SPIFFE ID",
+			code:   codes.InvalidArgument,
+			req: &svidpb.MintJWTSVIDRequest{
+				Audience: []string{"audience1", "audience2"},
+			},
+		},
+		{
+			name:   "request with empty SPIFFE ID",
+			err:    "spiffeid: trust domain is empty",
+			logMsg: "Failed to parse SPIFFE ID",
+			code:   codes.InvalidArgument,
+			req: &svidpb.MintJWTSVIDRequest{
+				Id:       &types.SPIFFEID{},
+				Audience: []string{"audience1", "audience2"},
+			},
+		},
+		{
+			name:   "request missing audience",
+			err:    "request must specify at least one audience",
+			logMsg: "Request must specify at least one audience",
+			code:   codes.InvalidArgument,
+			req: &svidpb.MintJWTSVIDRequest{
+				Id: &types.SPIFFEID{
+					TrustDomain: spiffeID.TrustDomain().String(),
+					Path:        spiffeID.Path(),
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			// Setup service
+			c.service.err = testCase.serviceErr
+			c.service.id = testCase.spiffeID
+			c.service.token = testCase.token
+			c.service.issuedAt = testCase.issuedAt
+			c.service.expiresAt = testCase.expiresAt
+
+			// Call MintJWTSVID
+			resp, err := c.svidClient.MintJWTSVID(ctx, testCase.req)
+
+			// Verify expected error
+			if testCase.err != "" {
+				spiretest.RequireGRPCStatusContains(t, err, testCase.code, testCase.err)
+				require.Nil(t, resp)
+				require.Equal(t, testCase.logMsg, c.logHook.LastEntry().Message)
+
+				return
+			}
+
+			// Verify response
+			require.NoError(t, err)
+			require.Equal(t, testCase.spiffeID.TrustDomain().String(), resp.Svid.Id.TrustDomain)
+			require.Equal(t, testCase.spiffeID.Path(), resp.Svid.Id.Path)
+			require.Equal(t, testCase.token, resp.Svid.Token)
+			require.Equal(t, testCase.issuedAt.UTC().Unix(), resp.Svid.IssuedAt)
+			require.Equal(t, testCase.expiresAt.UTC().Unix(), resp.Svid.ExpiresAt)
+
+			c.logHook.Reset()
+		})
+	}
+}
+
 type FakeService struct {
 	svid.Service
 
 	id        spiffeid.ID
 	certChain [][]byte
 	err       string
 	expiresAt time.Time
+	issuedAt  time.Time
+	token     string
```

Comment:
Hmm, wonder if we want to start separating these fields into per-rpc groups (maybe in their own structs). There would be some repetition but then it would be clear what fields are for which RPC...

Reply:
Yes, I've been debating with myself the best way to handle this. I have in mind an idea to have a better organization of the shared components across the `svid_test` package. I'll start by separating this in per-rpc structs and I can go from there when I reorganize the shared components for testing.